src/solvers/refinement/string_refinement_util.cpp

```diff
@@ -322,18 +322,22 @@ std::vector<mp_integer> string_insertion_builtin_functiont::eval(
   const std::vector<mp_integer> &args_value) const
 {
   PRECONDITION(args_value.size() >= 1 || args_value.size() <= 3);
-  const std::size_t &offset = numeric_cast_v<std::size_t>(args_value[0]);
-  const std::size_t &start =
-    args_value.size() > 1 ? numeric_cast_v<std::size_t>(args_value[1]) : 0;
-  const std::size_t &end = args_value.size() > 2
-                             ? numeric_cast_v<std::size_t>(args_value[2])
-                             : input2_value.size();
+  const auto offset = std::max<BigInt::llong_t>(args_value[0].to_long(), 0);
```

Comment:
I'd put this repeated "if less than zero then zero" into a static "diode" function. Only I'm not sure about the name.

Reply:
Don't use `BigInt` directly, use `mp_integer` instead.